src/vcpkg/dependencies.cpp

```diff
@@ -829,17 +826,19 @@ namespace vcpkg
                         if (!supports_expression.get()->evaluate(
                                 m_var_provider.get_dep_info_vars(spec.spec()).value_or_exit(VCPKG_LINE_INFO)))
                         {
-                            const auto msg = Strings::format("%s[%s] is only supported on '%s'",
-                                                             spec.port(),
-                                                             spec.feature(),
-                                                             to_string(*supports_expression.get()));
+                            auto localized_msg =
+                                msg::format(msgUnsupportedPortFeature,
+                                            msg::package_name = spec.port(),
+                                            msg::value = spec.feature(),
```

Comment:
I think you should change `UnsupportedPortFeature` to be:

```
"{spec} is only supported on '{supports_expression}'"
```

and then

```suggestion
                                            msg::spec = spec,
```

See https://github.com/microsoft/vcpkg-tool/blob/7d292feb55656816952b68efd4a1a738839d6c78/src/vcpkg/packagespec.cpp#L17-L21 and https://github.com/microsoft/vcpkg-tool/blob/7d292feb55656816952b68efd4a1a738839d6c78/include/vcpkg/packagespec.h#L222 which should have already taught fmt how to write these.